### PR TITLE
Fix SMS codes with leading zeros

### PR DESCRIPTION
**Pull request by joshbetz**
We're formatting the SMS code as a number, but it's really a string that just happens to be made up of numbers. If it starts with 0, we don't want the leading 0 to be truncated.

```
php > var_dump( sprintf( '%d', '08' ) );
string(1) "8"
```

```
php > var_dump( sprintf( '%s', '08' ) );
string(2) "08"
```

### CODE REVIEW

File: wpcom-vip-two-factor/sms-provider.php [hunk 1]
```diff
@@ -89,7 +89,7 @@ public function format_sms_message( $verification_code ) {
 		$parse = parse_url( home_url() );
 		$home_url_without_protocol = $parse['host'] . $parse['path'];
 
-		$format = '%1$d is your %2$s verification code.' . "\n\n" . '@%3$s #%1$d';
+		$format = '%1$s is your %2$s verification code.' . "\n\n" . '@%3$s #%1$s';
```
Comment (mjangda) [inline wpcom-vip-two-factor/sms-provider.php]: 💯 

We should probably add a test for this as well?

File: tests/wpcom-vip-two-factor/sms-provider.php [hunk 2]
```diff
@@ -7,15 +7,20 @@ public function setUp() {
...
 		$two_factor = Two_Factor_SMS::get_instance();
+		$actual   = $two_factor->format_sms_message( $token );
 
-		$token      = 123456;
-		$site_title = 'Test Blog'; // used set values for test site
-		$home_url   = 'example.org'; // used set values for test site
+		$this->assertEquals( $expected, $actual );
+	}
 
-		$format = '%1$d is your %2$s verification code.' . "\n\n" . '@%3$s #%1$d';
```
Comment (mjangda) [inline tests/wpcom-vip-two-factor/sms-provider.php]: Removed the logic from tests to keep things simple and explicit.

File: wpcom-vip-two-factor/sms-provider.php (hunk 1 above)
Comment (mjangda) [reply wpcom-vip-two-factor/sms-provider.php]: Added in 9594755

**Comment by hanifn** [comment]
r840-stacks

**Pull request merged**